### PR TITLE
Added support for replaced and overridden states to prefix lists module

### PR DESCRIPTION
**Pull request by stalabi1**
##### SUMMARY

I added support for replaced and overridden states to prefix lists module.
##### GitHub Issues
List the GitHub issues impacted by this PR. If no Github issues are affected, please indicate this with "N/A".

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

sonic_prefix_lists

##### OUTPUT
[regression-2023-05-12-09-33-28.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11466207/regression-2023-05-12-09-33-28.html.pdf)
[prefix_reg_output.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11466210/prefix_reg_output.log)

##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

### CODE REVIEW

File: plugins/modules/sonic_prefix_lists.py [hunk 1]
```diff
@@ -227,6 +231,95 @@
...
+#
+# Before state:
+# ------------
+#
+# sonic# show running-configuration ip prefix-list
+# !
+# ip prefix-list pfx1 seq 10 permit 1.2.3.4/24 ge 26 le 30
+# ip prefix-list pfx1 seq 20 deny 1.2.3.12/26
+# ip prefix-list pfx1 seq 30 permit 7.8.9.0/24
+#
```
Comment (kerry-meyer) [inline plugins/modules/sonic_prefix_lists.py:245]: Please modify this example to illustrate the fact that using "overidden" state will cause all prefix lists that aren't specified in the playbook to be deleted.

Adding one or two additional prefix lists (that get deleted) would show this clearly.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
The code, test cases, and test results all look good.

My only comment is to provide a more visible "Example" for overridden state to make it clear to users that this state causes deletion of any prefixes that are not specified in the playbook.

File: plugins/modules/sonic_prefix_lists.py (hunk 1 above)
Comment (stalabi1) [reply plugins/modules/sonic_prefix_lists.py:245]: Done

Comment (kerry-meyer) [reply plugins/modules/sonic_prefix_lists.py:245]: Thank you.

Resolved.

**Review by kerry-meyer — APPROVED** [review]
Thank you for adding the replaced and overridden support for the prefix lists resource module.

Approved.

**Pull request merged**